dht.go

```diff
@@ -187,16 +198,33 @@ func makeDHT(ctx context.Context, h host.Host, dstore ds.Batching, protocols []p
 		bucketSize:   bucketSize,
 	}
 
+	dht.setupEventSubscribers()
+
+	dht.ctx = dht.newContextWithLocalTags(ctx)
```

Comment:
Is this the right order? Subscribe then setup the context?

Reply:
event subscribers don't depend on the context, but logically it should be reversed, correct